### ISSUE TITLE
MacOS client crashes with NSInternalInconsistencyException

### ISSUE THREAD

**Issue by iammathew** [Xpra-org/xpra#3049]
Hello,

I currently try to get xpra working to connect from my MacBook (latest Mojave) to my Linux Workstation (Ubuntu 18.04 aka Bionic). Trying to run the example given on the website:

`xpra start ssh://mathew@<IP>/ --start=xterm`

I run into the following error:
```
2021-03-11 14:10:10.170 Xpra[47317:1339320] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff4d5429ad __exceptionPreprocess + 256
	1   libobjc.A.dylib                     0x00007fff77c66a17 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff4d55c50d -[NSException raise] + 9
	3   AppKit                              0x00007fff4ab015ca -[NSWindow(NSWindow_Theme) _postWindowNeedsToResetDragMarginsUnlessPostingDisabled] + 317
	4   AppKit                              0x00007fff4aafe9f7 -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 1479
	5   AppKit                              0x00007fff4aafe42a -[NSWindow initWithContentRect:styleMask:backing:defer:] + 45
	6   AppKit                              0x00007fff4adcad08 -[NSWindow initWithContentRect:styleMask:backing:defer:screen:] + 52
	7   libgdk-3.0.dylib                    0x000000010befed09 -[GdkQuartzNSWindow initWithContentRect:styleMask:backing:defer:screen:] + 137
	8   libgdk-3.0.dylib                    0x000000010bf11a61 _gdk_quartz_display_create_window_impl + 1137
	9   libgdk-3.0.dylib                    0x000000010bec0a05 _gdk_display_create_window_impl + 101
	10  libgdk-3.0.dylib                    0x000000010bedff96 gdk_window_new + 1638
	11  libgtk-3.0.dylib                    0x0000000110877942 gtk_window_realize + 1282
	12  libgobject-2.0.0.dylib              0x000000010acfee26 _g_closure_invoke_va + 198
	13  libgobject-2.0.0.dylib              0x000000010ad1584f g_signal_emit_valist + 1407
	14  libgobject-2.0.0.dylib              0x000000010ad16282 g_signal_emit + 130
	15  libgtk-3.0.dylib                    0x0000000110843f6f gtk_widget_realize + 575
	16  libgtk-3.0.dylib                    0x0000000110876bbb gtk_window_show + 123
	17  libgobject-2.0.0.dylib              0x000000010acfec3d g_closure_invoke + 189
	18  libgobject-2.0.0.dylib              0x000000010ad14b1c signal_emit_unlocked_R + 1212
	19  libgobject-2.0.0.dylib              0x000000010ad15b36 g_signal_emit_valist + 2150
	20  libgobject-2.0.0.dylib              0x000000010ad16282 g_signal_emit + 130
	21  libgtk-3.0.dylib                    0x00000001108431e7 gtk_widget_show + 391
	22  libgtk-3.0.dylib                    0x000000011050fb23 gtk_container_show_all + 211
	23  libgtk-3.0.dylib                    0x0000000110843a65 gtk_widget_show_all + 245
	24  libffi.7.dylib                      0x000000010ad669dd ffi_call_unix64 + 85
	25  ???                                 0x0000700015dfd480 0x0 + 123145669301376
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

Effectively crashing Xpra on the MacBook.
Versions used on workstation and mac are both 4.1.1!

Not sure what's causing this issue and could not find any related errors online.

Attaching full log:
```
(Xpra:47317): Gtk-CRITICAL **: 14:09:51.060: gtk_window_add_accel_group: assertion 'GTK_IS_WINDOW (window)' failed
2021-03-11 14:09:51,113 Xpra GTK3 client version 4.1.1 64-bit
2021-03-11 14:09:51,147  running on Mac OS X 10.14.6
2021-03-11 14:09:55,497 GStreamer version 1.18.3 for Python 3.9.2 64-bit
2021-03-11 14:09:55,505 Warning: socket directory '/Users/matthias/.xpra/clients'
2021-03-11 14:09:55,505  expected permissions 0o700 but found 0o755
2021-03-11 14:09:55,537 created unix domain socket '/Users/matthias/.xpra/clients/Matthiass-MacBook-Pro-2.local-47317'
2021-03-11 14:09:55,761 OpenGL_accelerate module loaded
2021-03-11 14:09:55,787 Using accelerated ArrayDatatype
2021-03-11 14:09:56,635 Warning: vendor 'Intel Inc.' is greylisted,
2021-03-11 14:09:56,636  you may want to turn off OpenGL if you encounter bugs
2021-03-11 14:09:56,667 OpenGL enabled with Intel(R) Iris(TM) Plus Graphics 645
2021-03-11 14:09:56,891 Connected (version 2.0, client OpenSSH_7.6p1)
2021-03-11 14:09:57,177 loaded RSA private key from '/Users/matthias/.ssh/id_rsa'
2021-03-11 14:09:57,628 Authentication (publickey) successful!
2021-03-11 14:09:58,446  SSH: 'Entering daemon mode; any further errors will be reported to:'
2021-03-11 14:09:58,446  SSH: '  /run/user/1000/xpra/S9336.log'
2021-03-11 14:10:00,639  SSH: 'Actual display used: :5'
2021-03-11 14:10:00,639  SSH: 'Actual log file name is now: /run/user/1000/xpra/:5.log'

(Xpra:47317): Gtk-CRITICAL **: 14:10:08.042: gtk_window_add_accel_group: assertion 'GTK_IS_WINDOW (window)' failed
2021-03-11 14:10:08,075  keyboard settings: layout=us
2021-03-11 14:10:08,112  desktop size is 1440x900 with 1 screen:
2021-03-11 14:10:08,112   Matthiass-MacBook-Pro-2.local (285x179 mm - DPI: 128x128) workarea: 1440x877 at 0x23
2021-03-11 14:10:08,112     monitor 2
2021-03-11 14:10:10.170 Xpra[47317:1339320] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff4d5429ad __exceptionPreprocess + 256
	1   libobjc.A.dylib                     0x00007fff77c66a17 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff4d55c50d -[NSException raise] + 9
	3   AppKit                              0x00007fff4ab015ca -[NSWindow(NSWindow_Theme) _postWindowNeedsToResetDragMarginsUnlessPostingDisabled] + 317
	4   AppKit                              0x00007fff4aafe9f7 -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 1479
	5   AppKit                              0x00007fff4aafe42a -[NSWindow initWithContentRect:styleMask:backing:defer:] + 45
	6   AppKit                              0x00007fff4adcad08 -[NSWindow initWithContentRect:styleMask:backing:defer:screen:] + 52
	7   libgdk-3.0.dylib                    0x000000010befed09 -[GdkQuartzNSWindow initWithContentRect:styleMask:backing:defer:screen:] + 137
	8   libgdk-3.0.dylib                    0x000000010bf11a61 _gdk_quartz_display_create_window_impl + 1137
	9   libgdk-3.0.dylib                    0x000000010bec0a05 _gdk_display_create_window_impl + 101
	10  libgdk-3.0.dylib                    0x000000010bedff96 gdk_window_new + 1638
	11  libgtk-3.0.dylib                    0x0000000110877942 gtk_window_realize + 1282
	12  libgobject-2.0.0.dylib              0x000000010acfee26 _g_closure_invoke_va + 198
	13  libgobject-2.0.0.dylib              0x000000010ad1584f g_signal_emit_valist + 1407
	14  libgobject-2.0.0.dylib              0x000000010ad16282 g_signal_emit + 130
	15  libgtk-3.0.dylib                    0x0000000110843f6f gtk_widget_realize + 575
	16  libgtk-3.0.dylib                    0x0000000110876bbb gtk_window_show + 123
	17  libgobject-2.0.0.dylib              0x000000010acfec3d g_closure_invoke + 189
	18  libgobject-2.0.0.dylib              0x000000010ad14b1c signal_emit_unlocked_R + 1212
	19  libgobject-2.0.0.dylib              0x000000010ad15b36 g_signal_emit_valist + 2150
	20  libgobject-2.0.0.dylib              0x000000010ad16282 g_signal_emit + 130
	21  libgtk-3.0.dylib                    0x00000001108431e7 gtk_widget_show + 391
	22  libgtk-3.0.dylib                    0x000000011050fb23 gtk_container_show_all + 211
	23  libgtk-3.0.dylib                    0x0000000110843a65 gtk_widget_show_all + 245
	24  libffi.7.dylib                      0x000000010ad669dd ffi_call_unix64 + 85
	25  ???                                 0x0000700015dfd480 0x0 + 123145669301376
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by totaam** [comment]
That's weird, I'm testing on the exact same version of MacOS and not seeing this crash.
Did earlier versions work on this system? (ie 4.0.x) Have you tried the 4.2 beta? https://xpra.org/beta/MacOS/

Can you try adding `--opengl=no --splash=no` to your command line to see if it helps?
If it doesn't, please add `-d all` so we can see what xpra is trying to do when the crash occurs.

**Comment by area42** [comment]
extra datapoint I have the same error.  I tried the options --opengl=no. I am using Big Sur 11.2.2 

[xpra.log](https://github.com/Xpra-org/xpra/files/6134513/xpra.log) 

Same result for version 4.2 
[xpra4.2.log](https://github.com/Xpra-org/xpra/files/6134569/xpra4.2.log)

**Comment by srcj** [comment]
Hi, 
I've got same error running xpra 4.1.1 using my macbook air as client and ubuntu 20.04 as server. 
The beta version gave the same error but when I tried version 4.0.6 there was no error and connection was successful.

**Comment by totaam** [comment]
I can reproduce the problem now, even on the original VM - as long as I move the build directory away.

So this looks like a packaging issue with the audio component.
Until I can fix this properly, you can use this temporary workaround:
```
./Xpra.app/Contents/MacOS/Xpra --speaker=no --microphone=no
```

**Comment by totaam** [comment]
OK, so the bug was in the notification handling. On my system the client shows a notification to complain about the missing audio devices, which is a "feature" of MacOS in VirtualBox.
What is the warning that you're seeing @area42, @srcj, @iammathew ? OpenGL? Audio?

The fix above allows the client to run again. There is a 4.2 beta build you can use here: https://xpra.org/beta/MacOS/

I will try to find a better fix so that on the platforms that can support it / need it, we can continue to show notifications before the client's main loop is actually running.

**Comment by totaam** [comment]
4.2 switches back to native notifications on MacOS and ensures we call from the main thread, without penalizing the other backends unnecessarily.
For safety, I've applied the original - more coarse - fix in 4.1.x
